### PR TITLE
Require an icon to be specified on icon mappings

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

In #1874, we introduced the ability to have icon mappings with an `undefined` icon to support use cases where the cell should be empty with non-empty group rows and where the cell should have only text. These use cases are now covered by usages of `nimble-mapping-empty` and `nimble-mapping-text`, both of which are supported by the mapping column. Therefore, we no longer need to support icon mappings without an icon.

## 👩‍💻 Implementation

Mostly reverting #1874, with a few minor updates to account for updates made in the source since the change was put it.

## 🧪 Testing

- Added new unit test that verifies setting an `undefined` icon on an icon mapping in the mapping column results in a validation error
- Updated matrix tests

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by mollykreis** [comment]
@m-akinc, will you buddy this PR for me?

**Review by m-akinc — APPROVED** [review]

File: change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "major",
+  "comment": "Require an icon to be specified on icon mappings",
```
[the remark this answers is not in the document]
Comment (mollykreis) [reply change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json]: Updated change description.

[the remark this answers is not in the document]
Comment (rajsite) [reply change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json]: I'd leave off the surrounding square brackets, a prefix of `Breaking change: ` seems sufficient.
Having a guideline to make sure breaking changes provide actionable steps / links to clear instructions for adopting the change sounds like a good idea 👍
Seems like the top-level Contributing topic on beachball is a good spot: https://github.com/ni/nimble/blob/main/CONTRIBUTING.md#beachball-change-file

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (mollykreis) [reply change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json]: Turns out we've already documented that guideline in the exact place @rajsite recommended 😄 

https://github.com/ni/nimble/blob/d4f54b4ba1399976395e9b6532229d1a81bd4335/CONTRIBUTING.md?plain=1#L85

**Review by jattasNI — APPROVED** [review]

File: change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajsite) [reply change/@ni-nimble-components-4d64effc-fc76-4e79-895f-8d61e2d7b02f.json]: Amazing lol

**Pull request merged**